src/core/decorators/singleton_decorator.py

```diff
@@ -0,0 +1,13 @@
+# structure for the singleton decorator
+
+
+def singleton(cls):
+
+    instance = {}
+
+    def wrapper(*args, **kwargs):
```

Comment:
Can we use  class name instead as the key? There is no need yo use the whole class as the key of the dict